### PR TITLE
[IIIF-1001] Fix NPE in Item

### PR DESCRIPTION
**Pull request by DRickard**
A braces-and-belt approach, checking file path at object creation and again in `getFile()`.
Updates:
* Updated `Item.java` to check file path at creation and set `hasImageFile` false when empty; added `if` clause in `getFile()` to double-check file path.
* Added test case to `ItemTest.java` to verify changes to `Item.java`.

### CODE REVIEW

File: src/main/java/edu/ucla/library/bucketeer/Item.java [hunk 1]
```diff
@@ -161,7 +164,9 @@ public Item setAccessURL(final String aAccessURL) {
         final String filePath;
         final File file;
 
-        if (!hasFile() || myFilePath.isEmpty()) {
+        if (myFilePath == null) {
```
Comment (ksclarke) [inline src/main/java/edu/ucla/library/bucketeer/Item.java:167]: Why not `if (myFilePath == null || !hasFile() || myFilePath.isEmpty()) {` but I'm thinking another way to handle this is to just do `myFilePath = Optional.empty()` in the default empty `public Item()` constructor. That would also address the issue with `getFilePath` returning a null instead of an empty optional, which I think is counter to what one would expect for an optional return type.

File: src/test/java/edu/ucla/library/bucketeer/ItemTest.java [hunk 2]
```diff
@@ -95,6 +95,17 @@ public final void testSetFilePath() throws IOException {
         assertEquals(filePath, item.setFilePath(Optional.of(filePath)).getFilePath().get());
     }
 
+    /**
+     * Tests setting an empty file path.
+     *
+     */
+    @Test
```
Comment (ksclarke) [inline src/test/java/edu/ucla/library/bucketeer/ItemTest.java:102]: We probably need a test for the default empty constructor too.

**Review by ksclarke — CHANGES_REQUESTED** [review]
`public Optional<String> getFilePath()` would still return a null, I believe, if the default empty constructor is used. It should return an `Optional.empty()`.

File: src/main/java/edu/ucla/library/bucketeer/Item.java [hunk 3]
```diff
@@ -62,6 +63,9 @@ public Item() {
     @JsonIgnore
     public Item(final String aID, final String aFilePath) {
         myFilePath = Optional.ofNullable(aFilePath);
+        if (myFilePath.isEmpty()) {
+            hasImageFile = false;
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/bucketeer/Item.java:67]: @ksclarke Would it make a difference if we set this member variable to false initially (L41) and set it to true here (`if (myFilePath.isPresent())`) instead?

Comment (markmatney) [reply src/main/java/edu/ucla/library/bucketeer/Item.java:67]: I guess my reason for asking is because a default of `true` there doesn't seem to match with the file path being empty

Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/Item.java:67]: Yes, I think that makes sense.

File: src/main/java/edu/ucla/library/bucketeer/Item.java [hunk 4]
```diff
@@ -164,9 +165,7 @@ public Item setAccessURL(final String aAccessURL) {
         final String filePath;
         final File file;
 
-        if (myFilePath == null) {
-            return Optional.empty();
-        } else if (!hasFile() || myFilePath.isEmpty()) {
+        if (myFilePath == null || !hasFile() || myFilePath.isEmpty()) {
```
Comment (ksclarke) [inline src/main/java/edu/ucla/library/bucketeer/Item.java:168]: Since the constructors both set `myFilePath` now, the only way this could be null is if someone passes a null to `setFilePath(final Optional<String> aFilePath)`. I'd be inclined to check for null in that setter rather than having to check for null here (that localizes the exception at the point of the problem (i.e. passing a null)). In the setter you could use `Objects.requireNonNull(aFilePath)` or the version that takes a string message. Then, here, you only have to check `hasFile()` and `myFilePath.isEmpty()`.

File: src/main/java/edu/ucla/library/bucketeer/Item.java (hunk 3 above)
Comment (markmatney) [reply src/main/java/edu/ucla/library/bucketeer/Item.java:67]: I think that would be good to change for readability, but not essential

**Review by ksclarke — APPROVED** [review]
:+1:

**Review by markmatney — APPROVED** [review]
